### PR TITLE
Streamline animation vars

### PR DESCRIPTION
**Pull request by spacek531**
renames some vars to be more descriptive
refactors steam spawning logic
shouldn't require network bump

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -7328,88 +7327,88 @@ void Vehicle::UpdateAdditionalAnimation()
...
                 animation_frame += 1;
                 animation_frame &= 7;
                 Invalidate();
             }
             break;
         case VEHICLE_ENTRY_ANIMATION_HELICARS: // loc_6D63F5
-            *curVar_C8 += _vehicleVelocityF64E08;
-            al = (*curVar_C8 >> 18) & 3;
-            if (animation_frame != al)
+            *curAnimation_state += _vehicleVelocityF64E08;
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp:7381]: `curAnimationState`.

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -7275,11 +7275,11 @@ void Vehicle::UpdateSpinningCar()
  */
 void Vehicle::UpdateAdditionalAnimation()
 {
-    uint8_t al{};
-    uint8_t ah{};
+    uint8_t target_frame{};
+    uint8_t steam_frame{};
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp:7279]:
```suggestion
    uint8_t targetFrame{};
    uint8_t steamFrame{};
```

File: src/openrct2/ride/Vehicle.h [hunk 3]
```diff
@@ -181,7 +181,7 @@ struct Vehicle : SpriteBase
     };
     uint8_t animation_frame;
     uint8_t pad_C6[0x2];
-    uint16_t var_C8;
+    uint16_t animation_state;
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.h:184]: We are in the process of renaming everything to `camelCase`. As such, the appropriate name would be `animationState`.

File: src/openrct2/ride/Vehicle.cpp [hunk 4]
```diff
@@ -7288,16 +7288,15 @@ void Vehicle::UpdateAdditionalAnimation()
...
-                animation_frame = ah;
-                al &= 0x02;
-                ah &= 0x02;
-                if (al != ah)
+            *curAnimation_state += _vehicleVelocityF64E08;
+            target_frame = (*curAnimation_state >> 20) & 3;
+            if (animation_frame != target_frame)
+            {
+                steam_frame = animation_frame;
+                animation_frame = target_frame;
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp:7296]: Are these two lines the refactor you’re talking about?

Comment (spacek531) [reply src/openrct2/ride/Vehicle.cpp:7296]: yes

**Comment by duncanspumpkin** [comment]
This is based off of work from here? https://github.com/OpenRCT2/OpenRCT2/pull/11269

**Comment by spacek531** [comment]
No, it's branched from develop. I want to rethink the animation speed modifier.

File: src/openrct2/ride/Vehicle.cpp [hunk 5]
```diff
@@ -7275,11 +7275,11 @@ void Vehicle::UpdateSpinningCar()
...
 void Vehicle::UpdateAdditionalAnimation()
 {
-    uint8_t al{};
-    uint8_t ah{};
+    uint8_t targetFrame{};
+    uint8_t curFrame{};
     uint32_t eax{};
 
-    uint32_t* curVar_C8 = reinterpret_cast<uint32_t*>(&var_C8);
+    uint32_t* curAnimationState = reinterpret_cast<uint32_t*>(&animationState);
```
Comment (ZehMatt) [inline src/openrct2/ride/Vehicle.cpp:7282]: This is horrible, its a cast from 16 bit to 32 bit. I know its been like this from before but this is surely wrong.

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp:7282]: Would need to introduce a union ideally

Comment (ZehMatt) [reply src/openrct2/ride/Vehicle.cpp:7282]: I think this needs some further looking into, its probably one field and not two.

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp:7282]: I'm pretty sure it's just one more union situation. There were lots of them originally and it seems this one was missed.

Comment (ZehMatt) [reply src/openrct2/ride/Vehicle.cpp:7282]: Some animation types have different shifts, I think making this 32 bit and then have the right masks is better than the union here.

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp:7282]: Let’s do this in another PR.

**Pull request merged**